UPGRADING.md

```diff
@@ -1,5 +1,18 @@
 # Upgrading Guide
 
+## Upgrading from `17.0.0` to `18.0.0`
```

Comment:
`core` is updated to `18.0.0` but `sns` and `amqp` are updated to different versions, should we indicate that here?

Reply:
I think breaking change is the same for all libraries, right?
Maybe we can do something like:
```
- mqt/core from 17.0.0 to 18.0.0 
- amqp from x to y 
- ...
```